include/re_main.h

```diff
@@ -35,10 +36,11 @@ typedef void (fd_h)(int flags, void *arg);
 typedef void (re_signal_h)(int sig);
 
 
-int   fd_listen(re_sock_t fd, int flags, fd_h *fh, void *arg);
-void  fd_close(re_sock_t fd);
+int   fd_alloc(struct re_fhs **fhs);
+int   fd_listen(struct re_fhs *fhs, re_sock_t fd, int flags, fd_h *fh,
+		void *arg);
```

Comment:
perhaps alloc and listen can be the same function?

perhaps fd_listen can take a double-pointer, and update the state if pointer is set ?